x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -244,9 +249,13 @@ func (k Keeper) SubmitTxsEpoch(
 		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to get epoch tracker for %s", epochType)
 	}
 	// BUFFER by 5% of the epoch length
-	bufferSize := cast.ToInt64(k.GetParam(ctx, types.KeyBufferSize))
-	BUFFER := epochTracker.Duration / bufferSize
-	timeoutNanos := epochTracker.NextEpochStartTime - BUFFER
+	bufferSizeParam := cast.ToInt64(k.GetParam(ctx, types.KeyBufferSize))
+	bufferSize := epochTracker.Duration / bufferSizeParam
+	timeoutNanos := epochTracker.NextEpochStartTime - bufferSize
+	// TODO safety: it's possible the cast below will handle this case with a graceful error, but leaving this here to make double sure until we can be sure
+	if timeoutNanos < 0 {
```

Comment:
timeoutNanos is absolute, not relative, so it shouldn't ever be l.t. zero

there's a check in the underlying `SendTx` that asserts that timeoutNanos is g.t. the current block time
```
if uint64(ctx.BlockTime().UnixNano()) >= timeoutTimestamp {
	return 0, icatypes.ErrInvalidTimeoutTimestamp
}
```

I think possible the check we want here is that bufferSize is less than one epoch and positive?

Reply:
timeoutNanos handled with To___E. 
Added the buffer check